src/tracer.h

```diff
@@ -40,6 +40,8 @@
 #include "opentelemetry/sdk/resource/resource.h"
 #include "opentelemetry/sdk/trace/processor.h"
 #include "opentelemetry/sdk/trace/simple_processor_factory.h"
```

Comment:
If we are removing `SimpleProcessFactory`, then it makes sense to remove this input